### PR TITLE
DA Verifier light client proof tests

### LINKED ISSUE #1020: Write BitcoinDA Verifier tests against light client namespace

### PR DESCRIPTION
**Pull request by yaziciahmet**
# Description

Added light client proof namespace tests to DAVerifier. I only added tests that are exclusive to light client proofs. For instance, I didn't add a new `missing_tx_in_completeness_proof_lcp` test, as the purpose of this test is ensure the inclusion and completeness proof txs match, irrelevant of their content.

## Linked Issues
- Fixes #1020 

## Testing
Describe how these changes were tested. If you've added new features, have you added unit tests?

## Docs
Describe where this code is documented. If it changes a documented interface, have the docs been updated?

### CODE REVIEW

File: crates/bitcoin-da/src/service.rs
Comment (eyusufatik) [inline crates/bitcoin-da/src/service.rs:1]: can you add extract_relevant test for light client namespace?

I found a bug there today :) 

solved it and committed to #1502

File: crates/bitcoin-da/src/helpers/test_utils.rs [hunk 1]
```diff
@@ -23,26 +30,50 @@ pub(crate) fn get_mock_txs() -> Vec<Transaction> {
...
+                    let blob = decompress_blob(&complete.body);
+                    (blob, complete.public_key, hash)
+                }
+                super::parsers::ParsedLightClientTransaction::Aggregate(aggregate) => {
+                    let hash = aggregate
+                        .get_sig_verified_hash()
+                        .expect("Invalid sighash on aggregate zk proof");
+                    (aggregate.body, aggregate.public_key, hash)
+                }
+                _ => unimplemented!(),
```
Comment (eyusufatik) [inline crates/bitcoin-da/src/helpers/test_utils.rs:65]: why unimplemented? we can get chunk in a block

**Review by eyusufatik — CHANGES_REQUESTED** [review]

File: crates/bitcoin-da/src/service.rs
Comment (eyusufatik) [reply crates/bitcoin-da/src/service.rs:1]: I meant #1491

File: crates/bitcoin-da/src/helpers/test_utils.rs (hunk 1 above)
Comment (yaziciahmet) [reply crates/bitcoin-da/src/helpers/test_utils.rs:65]: we should ignore chunked transactions in here as this function is used for retrieving relevant prefixed transactions, and requesting chunks is later going to be performed on da service.

this also means that we can't write a unit test currently for extract_relevant_zk_proof as it requires local running regtest bitcoin. I left it as ignored for now to have a reference when we carry the regtest needing tests to bitcoin-e2e.

**Review by eyusufatik — APPROVED** [review]
aproving on accounts that a follow up of #1503 pr will be implemented

**Pull request merged**